dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/Util.java

```diff
@@ -323,6 +354,24 @@ public static String names(Collection<Class<?>> classes) {
         return b.toString();
     }
 
+    /**
+     * List of names of repository method annotations that represent operations.
+     * Enclosed in brackets and delimited by comma.
+     *
+     * @param producer producer of the repository bean, from which it can be
+     *                     determined if the repository is stateful or stateless.
+     */
+    @Trivial
+    static String operationAnnoNames(RepositoryProducer<?> producer) {
+        Set<Class<? extends Annotation>> annoClasses = producer.provider().compat //
+                        .operationAnnoTypes(producer.stateful());
+
+        StringBuilder b = new StringBuilder('[');
+        for (Class<?> annoClass : annoClasses)
+            b.append(b.isEmpty() ? "" : ", ").append(annoClass.getSimpleName());
+        return b.append(']').toString();
```

Comment:
```suggestion
        return annoClasses.stream()
                           .map(c -> c.getSimpleName)
                           .collect(Collectors.joining(", ", "[", "]"));
```

issue: Since you instantiate the string builder with `'['` the `isEmpty()` check will always be false resulting in a string like, `[, a, b, c...`

Reply:
Nice catch on these. I added the brackets later and that breaks the condition in the loop.